include/irr/asset/ICPUPipelineCache.h

```diff
@@ -110,10 +110,21 @@ class ICPUPipelineCache final : public IAsset
 	size_t conservativeSizeEstimate() const override { return 0ull; /*TODO*/ }
 	void convertToDummyObject(uint32_t referenceLevelsBelowToConvert = 0u) override
 	{
-		if (canBeConvertedToDummy())
+		if (!isADummyObjectForCache() && canBeConvertedToDummy())
```

Comment:
as I wrote, you could fold `usADummyObjectForCache` into `canBeConvertedToDummy`